std/experimental/checkedint/traits.d

```diff
@@ -0,0 +1,261 @@
+
+    /// See $(REF _intFlagPolicyOf, std,experimental,checkedint,flags)
+    alias intFlagPolicyOf = ciFlags.intFlagPolicyOf;
+
+
+// checkedint ////////////////////////////////////////////
+static import chkd = std.experimental.checkedint;
+
+    /// See $(REF _isSafeInt, std,experimental,checkedint)
+    alias isSafeInt = chkd.isSafeInt;
+
+    /// See $(REF _isSmartInt, std,experimental,checkedint)
+    alias isSmartInt = chkd.isSmartInt;
+
+    /// See $(REF _isCheckedInt, std,experimental,checkedint)
+    alias isCheckedInt = chkd.isCheckedInt;
+
+    /// See $(REF _hasBitOps, std,experimental,checkedint)
+    alias hasBitOps = chkd.hasBitOps;
+
+    /// See $(REF _BasicScalar, std,experimental,checkedint)
+    alias BasicScalar = chkd.BasicScalar;
+
+
+// std.traits ////////////////////////////////////////////
+static import bsct = std.traits;
+
+    private template isEx(alias Predicate, T)
+    {
+        static if (isCheckedInt!T)
+            enum isEx = Predicate!(BasicScalar!T);
+        else
+            enum isEx = Predicate!T;
+    }
+
+    /// See $(REF isScalarType, std,traits)
+    alias isBasicScalar = bsct.isScalarType;
+    /// `checkedint`-aware wrapper for $(REF _isScalarType, std,traits)
+    template isScalarType(T)
+    {
+        alias isScalarType = isEx!(isBasicScalar, T);
+    }
+    ///
+    unittest
+    {
+        foreach (T; AliasSeq!(int, ushort, double, bool))
+            assert(isBasicScalar!T && isScalarType!T);
+
+        assert(!isBasicScalar!(SmartInt!int));
+        assert( isScalarType!(SmartInt!int));
+
+        foreach (T; AliasSeq!(int[]))
+            assert(!(isBasicScalar!T || isScalarType!T));
+    }
+
+    /// See $(REF isNumeric, std,traits)
+    alias isBasicNum = bsct.isNumeric;
+    /// `checkedint`-aware wrapper for $(REF _isNumeric, std,traits)
+    template isNumeric(T)
+    {
+        alias isNumeric = isEx!(isBasicNum, T);
+    }
+    ///
+    unittest
+    {
+        foreach (T; AliasSeq!(int, ushort, double))
+            assert(isBasicNum!T && isNumeric!T);
+
+        assert(!isBasicNum!(SmartInt!int));
+        assert( isNumeric!(SmartInt!int));
+
+        foreach (T; AliasSeq!(int[], bool))
+            assert(!(isBasicNum!T || isNumeric!T));
+    }
+
+    /// See $(REF _isFloatingPoint, std,traits)
+    alias isFloatingPoint = bsct.isFloatingPoint;
+/+
+    /// See $(REF isFixedPoint, std,traits)
+    alias isBasicFixed = bsct.isFixedPoint;
+    /// `checkedint`-aware wrapper for $(REF isFixedPoint, std,traits)
+    template isFixedPoint(T)
+    {
+        alias isFixedPoint = isEx!(isBasicFixed, T);
+    }
+    ///
+    unittest
+    {
+        foreach (T; AliasSeq!(int, ushort, bool))
+            assert(isBasicFixed!T && isFixedPoint!T);
+
+        assert(!isBasicFixed!(SmartInt!int));
+        assert( isFixedPoint!(SmartInt!int));
+
+        foreach (T; AliasSeq!(double, int[]))
+            assert(!(isBasicFixed!T || isFixedPoint!T));
+    }
++/
```

Comment:
It's a `checkedint`-aware version of `isFixedPoint`, which is currently a `private` helper in `checkedint`'s `package.d`, but which I would like to add to `std.traits`. Search `package.d` for:
```D
// maybe add these to std.traits? ///////////////////////////
```